src/pytest_cov/compat.py

```diff
@@ -31,6 +34,21 @@ def testsfailed(self, value):
         setattr(self._session, self._attr, value)
 
 
+if PYTEST_VERSION >= (3, 8):
+    def warn(config, message, category, stacklevel=1):
+        return warnings.warn(
+            message=message,
+            category=category,
+            stacklevel=stacklevel+1,
+        )
+else:
+    def warn(config, message, category, stacklevel=1):
+        return config.warn(
+            code=category.code,
+            message=message,
```

Comment:
@nicoddemus I'm not sure if this will now break fslocation should I add something like:
```suggestion
            message=message,
            fslocation=_pytest._code.getfslineno(sys._getframe(stacklevel)),
```

Reply:
otherwise I could convert this compat functions to return functools.partial objects, and be called like this:

```python
fn = compat.warn(...)
fn.func(*fn.args, **fn.kwargs)
```

Reply:
The fslocation is sorta useless since it only applies to ancient pytest no?